src/core/interpreter/index.ts

```diff
@@ -1140,7 +1140,34 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
                         signature: signature,
                     });
                     try {
-                        const returnValue = callee.call(this, ...args);
+                        let returnValue = callee.call(this, ...args);
+                        if (
+                            signature.returns !== ValueKind.Void &&
+                            signature.returns !== ValueKind.Dynamic &&
+                            returnValue.kind === ValueKind.Invalid
+                        ) {
+                            // When a function has a typed return, but no return statement is hit, Roku returns zero by default
+                            const coercedValue = tryCoerce(new Int32(0), signature.returns);
+                            if (coercedValue) {
+                                returnValue = coercedValue;
+                            } else {
+                                // When the typed return is not numeric, Roku raises a type mismatch error
+                                this.addError(
+                                    new TypeMismatch({
+                                        message: `Unable to cast`,
+                                        left: {
+                                            type: signature.returns,
+                                            location: this.location,
+                                        },
+                                        right: {
+                                            type: ValueKind.Int32,
+                                            location: this.location,
+                                        },
```

Comment:
perhaps left/right in the opposite order? 
Trying to cast `<bogus Integer>` to `<declared return type>`